Tensile/KernelWriterAssembly.py

```diff
@@ -1229,6 +1229,273 @@ def functionPrefix(self, kernel):
+                    """
+                    ignore this, not quite correct for mixed precision
+                    D.f[31:16] = S0.f[31:16] * S1.f[31:16] + S2.f[31:16]
+                    D.f[15:00] = S0.f[15:00] * S1.f[15:00] + S2.f[15:00]
+                    C[0] = A[0]*B[0]+D[0]
+                    C[1] = A[1]*B[1]+D[1]
+                    """
+              else:
+                b = blockB*2
+                a = blockA*2
+                for iui in range(0, innerUnroll):
+                  cStr = "v[%s+%u+%u*%u+0]" % ("vgprValuC", blockA, blockB, kernel["ThreadTile0"]) # /2 b/c of 2 f16's per 32-bit vgpr
+                  aStr = "v[%s+%u]" \
+                      % ("vgprValuA_X%u_I%u"%(m,iui), blockA)
+                  bStr = "v[%s+%u]" \
+                      % ("vgprValuB_X%u_I%u"%(m,iui), blockB)
+                  kStr += "v_pk_fma_f16 %s, %s, %s, %s op_sel:[0,0,0] op_sel_hi:[1,0,1]%s" % (cStr, aStr, bStr, cStr, self.endLine)
+
+                  cStr = "v[%s+%u+%u*%u+%u]" % ("vgprValuC", blockA, blockB, kernel["ThreadTile0"], kernel["ThreadTile0"]/2)
+                  kStr += "v_pk_fma_f16 %s, %s, %s, %s op_sel:[0,1,0] op_sel_hi:[1,1,1]%s" % (cStr, aStr, bStr, cStr, self.endLine)
+                  """
+                  D.f[31:16] = S0.f[31:16] * S1.f[31:16] + S2.f[31:16]
+                  D.f[15:00] = S0.f[15:00] * S1.f[15:00] + S2.f[15:00]
+                  C[0] = A[0]*B[0]+D[0]
+                  C[1] = A[1]*B[1]+D[1]
+                  """
+            elif self.version == (9,0,6):
+              if kernel["ProblemType"]["HighPrecisionAccumulate"]:
+                # we treat HighPrecisionAccumulate as expanded packed math
+                b = blockB*2
+                a = blockA*2
+                if kernel["LocalDotLayout"] > 1:    # Only supports LocalDotLayout == 2 for now
+                  lcldot = kernel["LocalDotLayout"]
+                  iua = blockA / ((kernel["ThreadTileA"]/2) / lcldot)
+                  iub = blockB / ((kernel["ThreadTileB"]/2) / lcldot)
+                  rema = blockA % ((kernel["ThreadTileA"]/2) / lcldot)
+                  remb = blockB % ((kernel["ThreadTileB"]/2) / lcldot)
+                  #print "lcldot %u, blockA %u, blockB %u, rema %u, remb %u, ThreadTileA %u%s" % (lcldot, blockA, blockB, rema, remb, kernel["ThreadTileA"], self.endLine)
+                  cStr = "v[%s+%u*2+%u*%u*2+0*2+0]" % ("vgprValuC", blockA, blockB, kernel["ThreadTile0"]) # *2 b/c of fp32
+                  cidx = blockA*2 + blockB*kernel["ThreadTile0"]*2 + 0
+                  aStr = "v[%s+%u]" \
+                      % ("vgprValuA_X%u_I%u"%(m,iua), rema*lcldot)
+                  bStr = "v[%s+%u]" \
+                      % ("vgprValuB_X%u_I%u"%(m,iub), remb*lcldot)
+                  kStr += "v_dot2_f32_f16 %s, %s, %s, %s //ValuC[%u] iua=%u iub=%u%s" % (cStr, aStr, bStr, cStr, cidx, iua, iub, self.endLine)
```

Comment:
Argh. Somehow my fixes for op_sel and op_sel_hi got missed. I see that locally too. Will fix after the merge, unless you want to do it now. Ought to be "op_sel:[0,0] op_sel_hi:[1,1]" for all dot2 instructions.